### PR TITLE
[FIX] No only search in the initial picking type. Also look without it

### PR DESCRIPTION
**Pull request by hbto**
Let it be Product A with components (BoM) Product B & Product C:
Bom for Product A is set to be using Picking Type A,
In the other hand Product B is with components (BoM) Product D & Product E:
Bom for Product B is set to be using Picking Type B,

When sending to produce Product A, it only takes into account Product A's picking type
and so method `_get_matching_bom` won't yield any BoM for Product B, because it only uses Product A's picking type.

This PR intend to use the default method so far, and if not result is returned it tries to look
for a result by no using the picking type at all

### CODE REVIEW

**Comment by hbto** [comment]
An Issued has been submitted to odoo https://github.com/odoo/odoo/issues/28086

**Comment by hbto** [comment]
@nhomar @moylop260

File: diet_line/models/procurement.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
```
Comment (moylop260) [inline diet_line/models/procurement.py]: Remove the coding for py3.x or odoo>=11.0

**Comment by moylop260** [comment]
Is not an issue for odoo too?

**Pull request merged**